### ISSUE TITLE
Aggregate on empty input returns no rows

### ISSUE THREAD

**Issue by timzimmermann** [hyrise/hyrise#869]
Currently, the Aggregate operator returns no rows if its input is empty, e.g. because of a TableScan matching no rows.

I'm not quite sure what the standard says, but in postgres, things are different.
- `select count(*) from t where s = 'does not exist'` returns one row with value 0
- `select sum(a) from t where s = 'does not exist'` returns one row with value `null`

Are we aware / have we thought about how to handle that / do we knowingly deviate from postgres?

**Comment by mrks** [comment]
Good catch. I believe that this was not a conscious decision. From a quick scroll over the code, my guess is that this for-loop does not get executed for an empty table and thus the needed data structures are not filled:

https://github.com/hyrise/hyrise/blob/144a41cf6e029a3e4350bc73ee1357935bdc5474/src/lib/operators/aggregate.cpp#L414

Postgres is right.
